src/ShallowWaterIMEX.jl

```diff
@@ -0,0 +1,218 @@
+function shallow_water_imex_time_step!(
+     h₂, u₂, uₚ, ϕ, F, q₁, q₂,                               # in/out args
+     H1h, H1hchol, h_wrk, u_wrk, A_wrk, Bchol,               # more in/out args
+     model, dΩ, dω, V, P, Q, R, S, f, g, h₁, u₁, u₀,         # in args
+     RTMMchol, L2MMchol, RTMM, L2MMinvD, dt, τ, leap_frog)   # more in args
+
+  # energetically balanced implicit-explicit second order shallow water solver
+  # reference: eqns (31-33) of
+  # https://github.com/BOM-Monash-Collaborations/articles/blob/main/energetically_balanced_time_integration/EnergeticallyBalancedTimeIntegration_SW.tex
+  #
+  # f          : coriolis force (field)
+  # g          : gravity (constant)
+  # h₁         : fluid depth at current time level
+  # u₁         : fluid velocity at current time level
+  # hₘ         : fluid depth at previous time level (for leap-frogging the first step)
+  # uₘ         : fluid velocity at previous time level (for leap-frogging the first step)
+  # RTMM       : H(div) mass matrix, ∫β⋅βdΩ, ∀β∈ H(div,Ω)
+  # L2MM       : L² mass matrix, ∫γγdΩ, ∀γ∈ L²(Ω)
+  # dt         : time step
+  # leap_frog  : do leap frog time integration for the first step (boolean)
+  # dΩ         : measure of the elements
+
+  n = get_normal_vector(model)
+  # explicit step for provisional velocity, uₚ
+  dt1 = dt
+  if leap_frog
+    dt1 = 2.0*dt
+  end
+
+  Fv,h₁v,h₂v = get_free_dof_values(F,h₁,h₂)
```

Comment:
@cbladwell @davelee2804 ... note that I have introduced a new method for the multi-method `get_free_dof_values` in `Helpers.jl`. Using this method, we just call `get_free_dof_values` once in the preamble. 

@cbladwell ... can you please give us a hand and apply the same pattern in the other two solvers as per required? 

Thanks!

Reply:
Addressed here ad29e55e1673461bdba7859ddaa294a4594e9d51